src/state/GlobalStore.js

```diff
@@ -124,7 +124,18 @@ class GlobalStore {
       'updateProgressByPage',
       (global, dispatch, pageId, qId, answered) => {
         const { investigationProgress: prevIPS } = global;
-        const { questions, answers: prevAnswers } = prevIPS[pageId];
+        const prevPageIPS = prevIPS[pageId];
+
+        if (!prevIPS[pageId]) {
+          return global;
+        }
+
+        const { questions, answers: prevAnswers } = prevPageIPS;
+
+        if (!questions[qId]) {
```

Comment:
On the page `/observable-universe/exploring-the-deep/`, I'm getting an error in the console (page not loading futher) that says, "Cannot read property 'randomGalaxies' of null" that points to this line.

Reply:
I'm not able to reproduce the issues you're describing.  Are you on the latest commit? Can you provide instructions to reproduce `"Cannot read property 'randomGalaxies' of null"` and `/observable-universe/exploring-the-deep-1/ is 'stuck' loading the page. Only the spinner is loading on the selector.`?

Reply:
No specific instructions other than go to the page. I did nothing other than clear my local storage.